### PR TITLE
[adsp] change to new addon to kodi callback way

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This is the first major change to become multiple instances to work.

There are the callbacks moved to the CActiveAEDSPAddon class. Reason is
to have this parts together and later with multiple instances to call
always right related CActiveAEDSPAddon class.

On addon is currently nothing needed to change, only a recompile to
match them.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]
Looking good.

**Comment by ksooo** [comment]
Of course you have on your radar that all adsp addons need an API version bump, addon version bump and a changelog update. ;-)

**Comment by AlwinEsch** [comment]
;-) adsp currently not working and have no changelog, but version is updated :-D

@AchimTuran does a rework to become them good.

**Comment by AchimTuran** [comment]
Still working on it. But so far no real add-on support. :b

**Comment by AchimTuran** [comment]
But it's great that you were able to merge it. We will improve it once ADSP V2 is ready to go.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_adsp_types.h [hunk 1]
```diff
@@ -465,6 +457,35 @@ extern "C" {
...
+
+    ADSPHANDLE (*SoundPlay_GetHandle)(void* kodiInstance, const char *filename);
+    void (*SoundPlay_ReleaseHandle)(void* kodiInstance, ADSPHANDLE handle);
+    void (*SoundPlay_Play)(void* kodiInstance, ADSPHANDLE handle);
+    void (*SoundPlay_Stop)(void* kodiInstance, ADSPHANDLE handle);
+    bool (*SoundPlay_IsPlaying)(void* kodiInstance, ADSPHANDLE handle);
+    void (*SoundPlay_SetChannel)(void* kodiInstance, ADSPHANDLE handle, AE_DSP_CHANNEL channel);
+    AE_DSP_CHANNEL (*SoundPlay_GetChannel)(void* kodiInstance, ADSPHANDLE handle);
+    void (*SoundPlay_SetVolume)(void* kodiInstance, ADSPHANDLE handle, float volume);
+    float (*SoundPlay_GetVolume)(void* kodiInstance, ADSPHANDLE handle);
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_adsp_types.h]: Just as info this Sound Play become removed in furture and separated to independent header to have on all addons (a bit reworked).

Comment (AchimTuran) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_adsp_types.h]: +1
That's good. Go for it.

**Pull request merged**